src/sonic-eventd/rsyslog_plugin/syslog_parser.cpp

```diff
@@ -10,19 +11,59 @@
  *
 */
 
-bool SyslogParser::parseMessage(string message, string& eventTag, event_params_t& paramMap) {
+bool SyslogParser::parseMessage(string message, string& eventTag, event_params_t& paramMap, lua_State* luaState) {
     for(long unsigned int i = 0; i < m_regexList.size(); i++) {
         smatch matchResults;
         vector<string> params = m_regexList[i]["params"];
-        if(!regex_search(message, matchResults, m_expressions[i]) || params.size() != matchResults.size() - 1) {
+        if(!regex_search(message, matchResults, m_expressions[i]) || params.size() != matchResults.size() - 1 || matchResults.size() < 4) {
             continue;
         }
+
+        if(!matchResults[1].str().empty() && !matchResults[2].str().empty() && !matchResults[3].str().empty()) { // found timestamp components
```

Comment:
Else for this if also means missing timestamp warrants an error message.
You may declare the formattedtimestamp string outside and write error message if empty covering both missing & invalid cases.

Reply:
Done